sentry/src/main/java/io/sentry/util/CollectionUtils.java

```diff
@@ -120,6 +120,23 @@ public static int size(final @NotNull Iterable<?> data) {
     return mappedList;
   }
 
+  /**
+   * Returns a new list which entries match a predicate specified by a parameter.
+   *
+   * @param predicate - the predicate
+   * @return a new list
+   */
+  public static @NotNull <T> List<T> filterListEntries(
+      final @NotNull List<T> list, final @NotNull Predicate<T> predicate) {
+    final List<T> filteredList = new ArrayList<>();
```

Comment:
`l`: given that we usually just just remove a few items it could make sense to init the new list with the size of the original one, like this
```suggestion
    final List<T> filteredList = new ArrayList<>(list.size());
```